src/nvim/lua/treesitter.c

```diff
@@ -1287,40 +1287,63 @@ static struct luaL_Reg querycursor_meta[] = {
   { NULL, NULL }
 };
 
+static inline bool set_uint32(lua_State *L, int idx, char const *name, uint32_t *res)
+  FUNC_ATTR_ALWAYS_INLINE FUNC_ATTR_NONNULL_ALL
+{
+  lua_getfield(L, idx, name);
+
+  if (!lua_isnil(L, -1)) {
+    int64_t value = lua_tointeger(L, -1);
+    if (value >= 0 && value <= UINT32_MAX) {
+      *res = (uint32_t)value;
+      return true;
+    }
+  }
+  lua_pop(L, 1);
+
+  return false;
+}
+
 int tslua_push_querycursor(lua_State *L)
 {
   TSNode node = node_check(L, 1);
 
   TSQuery *query = query_check(L, 2);
   TSQueryCursor *cursor = ts_query_cursor_new();
-  ts_query_cursor_exec(cursor, query, node);
```

Comment:
Why was `ts_query_cursor_exec()` called before the range, depth and limit were set?